public/app/features/query/components/CreateActionComponent.ts

```diff
@@ -0,0 +1,11 @@
+export type CreateActionComponent = React.ComponentType;
+
+let extraRenderCreateActions: CreateActionComponent[] = [];
+
+export function addExtraRenderCreateAction(extra: CreateActionComponent) {
+  extraRenderCreateActions = extraRenderCreateActions.concat(extra);
+}
+
+export function getAllExtraRenderCreateAction(): CreateActionComponent[] {
+  return extraRenderCreateActions;
+}
```

Comment:
I think we should mark this as internal
https://github.com/grafana/grafana/blob/0383becc46d9e1171919b05ecae0e798a7d202c0/public/app/features/dashboard/state/DashboardModel.ts#L254